SA3/qa_web/static/js/base.js

```diff
@@ -1,7 +1,59 @@
 /*JAVASCRIPT for QA_Web*/
 
+// Callback for voting, sends ajax request to server then updates the corresponding score
+function vote_callback(event){
+  var vote_button = event.currentTarget
+  $.ajax({
+    url: '/vote/',
+    method: "POST",
+    data: {
+      'button' : vote_button.id,
+    },
+    dataType: 'json',
+    success: function(data){
+      $('#' + data.id).text(data.new_score);
+    }
+  });
+}
+
+// From Django docs on CSRF
+function getCookie(name) {
+  var cookieValue = null;
+  if (document.cookie && document.cookie !== '') {
+      var cookies = document.cookie.split(';');
+      for (var i = 0; i < cookies.length; i++) {
+          var cookie = jQuery.trim(cookies[i]);
+          if (cookie.substring(0, name.length + 1) === (name + '=')) {
+              cookieValue = decodeURIComponent(cookie.substring(name.length + 1));
+              break;
+          }
+      }
+  }
+  return cookieValue;
+}
+
 $(document).ready(function(){
   var converter = Markdown.getSanitizingConverter();
   var editor = new Markdown.Editor(converter);
   editor.run();
+
+  // Add CSRF Token to ajax requests using POST
+  var csrftoken = getCookie('csrftoken')
+  $.ajaxSetup({
+    beforeSend: function(xhr, settings) {
+        if (settings.type == "POST" && !this.crossDomain) {
+            xhr.setRequestHeader("X-CSRFToken", csrftoken);
+        }
+    }
+  });
```

Comment:
Our site is now protected against cross-site attacks. Locally. Nice. I'm kidding.